dashboard/app/models/ability.rb

```diff
@@ -136,9 +136,24 @@ def initialize(user)
         # 1) the user is the project owner
         # 2) the user is the teacher of the project owner
         # 3) the user and the project owner are in the same code reivew group
-        project.owner.id == user.id ||
-          project.owner.student_of?(user) ||
-          (project.owner.code_review_groups & user.code_review_groups).any?
+        project.owner.id == user.id || can?(:code_review, project.owner)
+      end
+
+      # A user can review the code or another user if they are the user's teacher or if
+      # they're in a shared section with code review turned on and they're in the same code review group
+      can :code_review, User do |other_user|
+        return true if other_user.student_of?(user)
+
+        in_shared_section_with_code_review = user.shared_sections_with?(other_user).any?(&:code_review_enabled?)
```

Comment:
(very minor, up to you) I'm wondering if this would be more readable without the extra variable (since the functions are all well-named).

Reply:
I'm going to leave it, it was just way too long of a line otherwise